app/code/Magento/MediaContent/Model/GetAssetIdByContentFieldComposite.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContent\Model;
```

Comment:
Please move this class to API module and use the class for DI configuration as interfaces cannot have constructors declared and we cannot guarantee that configured argument is going to be accepted by the interface implementation
```suggestion
namespace Magento\MediaContentApi\Model;
```